File: backend/src/Models/GuideImmigration.js
const mongoose = require("mongoose");

const EtapeSchema = new mongoose.Schema({
  ordre: { type: Number, required: true },
  titre: { type: String, required: true },
  description: { type: String, required: true },
  emoji: { type: String, default: "📌" },
  duree: { type: String },
  cout: { type: String },
  documents: [{ type: String }]
});

const GuideImmigrationSchema = new mongoose.Schema({
  paysOrigine: { type: String, required: true },
  paysDestination: { type: String, required: true },
  titre: { type: String, required: true },
  description: { type: String, required: true },
  dureeTotal: { type: String, required: true },
  coutTotal: { type: String, required: true },
  etapes: [EtapeSchema]
});

module.exports = mongoose.model("GuideImmigration", GuideImmigrationSchema);

File: frontend/app/src/main/java/ca/uqac/vistudia/Models/GuideImmigration.kt
package ca.uqac.vistudia.Models

data class Etape(
    val ordre: Int,
    val emoji: String,
    val titre: String,
    val description: String,
    val duree: String?,
    val cout: String?,
    val documents: List<String>
)

data class GuideImmigrationItem(
    val _id: String,
    val paysOrigine: String,
    val paysDestination: String,
    val titre: String,
    val description: String,
    val dureeTotal: String,
    val coutTotal: String
)

data class GuideImmigrationDetail(
    val _id: String,
    val paysOrigine: String,
    val paysDestination: String,
    val titre: String,
    val description: String,
    val dureeTotal: String,
    val coutTotal: String,
    val etapes: List<Etape>
)